src/nupic/support/decorators.py

```diff
@@ -195,8 +194,8 @@ def myFunctionFoo():
   assert isinstance(retryExceptions, tuple), (
     "retryExceptions must be tuple, but got %r") % (type(retryExceptions),)
   
-  if getLoggerCallback is None:
-    getLoggerCallback = logging.getLogger
+  if logger is None:
+    logger = logging.getLogger(__name__)
```

Comment:
From https://docs.python.org/2/library/logging.html: "Multiple calls to getLogger() with the same name will always return a reference to the same Logger object."